### DISCUSSION TITLE
Cypress 10.3 cypress doesnt find all contains

### DISCUSSION THREAD

**Discussion by Dream-Work** [cypress-io/cypress#22716] [Questions and Help]
Hello,

Website: https://www.tagesschau.de

my code: cy.get('a').contains('Impressum').should('be.visible');

wont work, because it finds the hidden match but not the match at the bottom

Cypress found 1 (one) match, but the website has 2 matches. One is hidden in the top right corner burger menu under "Mehr".
The second match is found at the bottom.

Why is this happening?

thx Sven

**Comment by jjhelguero** [comment]
[`.contains()`](https://docs.cypress.io/api/commands/contains#Single-Element) will find the first element matching within the scope. In your instance it is finding the first one within your your returned list of `<a>` elements. You'll have to change your commands a bit to get the visible one of the two.

```js
cy.get('a')
  // find both matches
  .find('Impressum')
  // filter by visible
  .filter(':visible')
  .should('be.visible')
```

You can also shorten the commands by using the `:visible` selector within the `.find()`
```js
cy.get('a')
  // find visible matches
  .find('Impressum:visible')
  .should('be.visible')
```
